core/src/main/java/tc/oc/pgm/shops/ShopModule.java

```diff
@@ -0,0 +1,235 @@
+
+        if (location.isEmpty()) {
+          throw new InvalidXMLException("shopkeeper must have a location defined", shopkeeper);
+        }
+
+        Shop shop = shops.get(shopId);
+
+        if (shop == null) {
+          throw new InvalidXMLException(
+              "No shop with id '" + shopId + "' could be found", shopkeeper);
+        }
+
+        keepers.add(new ShopKeeper(name, location, mob, shop));
+      }
+
+      return shops.isEmpty() ? null : new ShopModule(shops, keepers);
+    }
+  }
+
+  private static List<Icon> parseIcons(
+      Element category, KitParser kits, ActionParser actions, FilterParser filters, Logger logger)
+      throws InvalidXMLException {
+    List<Icon> icons = Lists.newArrayList();
+    for (Element icon : XMLUtils.getChildren(category, "item")) {
+      icons.add(parseIcon(icon, kits, actions, filters, logger));
+    }
+
+    if (icons.size() > Category.MAX_ICONS) {
+      throw new InvalidXMLException(
+          "Categories may only contain up " + Category.MAX_ICONS + " icons", category);
+    }
+
+    if (icons.isEmpty()) {
+      throw new InvalidXMLException("At least one icon is required per category", category);
+    }
+
+    return icons;
+  }
+
+  private static Icon parseIcon(
+      Element icon, KitParser kits, ActionParser actions, FilterParser filters, Logger logger)
+      throws InvalidXMLException {
+
+    List<Payment> payments = Lists.newArrayList();
+
+    for (Element payment : XMLUtils.getChildren(icon, "payment")) {
+      payments.add(parsePayment(payment, kits));
+    }
+
+    if (payments.isEmpty()) {
+      payments.add(parsePayment(icon, kits));
+    }
+
+    ItemStack item = kits.parseItem(icon, false);
+    Filter filter = filters.parseFilterProperty(icon, "filter", StaticFilter.ALLOW);
+
+    Action<? super MatchPlayer> action;
+    Node actionNode = Node.fromAttr(icon, "action", "kit");
+    if (actionNode != null) {
+      action = actions.parseReference(actionNode, null, MatchPlayer.class);
+    } else {
+      // TODO: extract to a proper give-item action, handle full inventory situation.
+      action =
+          new AbstractAction<MatchPlayer>(MatchPlayer.class) {
+            @Override
+            public void trigger(MatchPlayer player) {
+              Map<Integer, ItemStack> overflow = player.getInventory().addItem(item);
+
+              // Drop extra items at feet
+              if (!overflow.isEmpty()) {
+                // will be translatable in production
+                player.sendWarning(
+                    text(
+                        "Purchase could not fit in your inventory, so it's been dropped at your feet"));
+                World world = player.getWorld();
+                overflow
+                    .values()
+                    .forEach(stack -> world.dropItemNaturally(player.getLocation(), stack));
+              }
+            }
+          };
```

Comment:
Given that we now have #1049 , this could be using an item kit with repair-tools, deduct-tools, and deduct-items all set to false.
That still doesn't help with what-if you don't have space in your inventory to buy the item, but *maybe* that should be a separate thing? (eg: warning the player their inventory is full if they try to buy something they don't have space for?)